File: app/src/main/java/com/example/boxcounter/data/local/ShiftDao.java
package com.example.boxcounter.data.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.boxcounter.model.entity.Shift;

import java.util.List;

@Dao
public interface ShiftDao {

    @Insert
    long insert(Shift shift);

    @Update
    void update(Shift shift);

    @Query("SELECT * FROM shifts WHERE active = 1 LIMIT 1")
    Shift getActiveShift();

    @Query("SELECT * FROM shifts ORDER BY startTime DESC")
    LiveData<List<Shift>> getAllShifts();

    @Query("UPDATE shifts SET active = 0 WHERE active = 1")
    void closeAllActive();
}

File: wear/src/main/java/com/boxcounter/wear/data/ShiftDao.kt
package com.boxcounter.wear.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import com.boxcounter.core.entity.Shift

@Dao
interface ShiftDao {

    @Query("SELECT * FROM shifts WHERE active = 1 LIMIT 1")
    suspend fun getActiveShift(): Shift?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertShift(shift: Shift)

    @Update
    suspend fun updateShift(shift: Shift)

    @Query("DELETE FROM shifts")
    suspend fun deleteAll()
}